src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -119,3 +533,1006 @@ async fn test_decider_installed() {
+        &mut client,
+        r#"
+            (seq
+                (call relay ("decider" "get_u32") ["counter"] counter)
+                (call relay ("decider" "list_get_strings") ["joined_deals"] deals)
+            )
+        "#,
+        "counter deals",
+        hashmap! {},
+    )
+    .await
+    .unwrap();
+
+    // Check that decider was really run
+    let counter = serde_json::from_value::<U32Value>(result.remove(0)).unwrap();
+    assert!(
+        !counter.absent,
+        "decider hasn't started yet (no counter in kv)"
+    );
+    assert_eq!(counter.num, 1, "decider wasn't run");
+
+    // Analyse joined deals
+    let deal = {
+        let mut deals = parse_joined_deals(result.remove(0));
+        assert_eq!(deals.len(), 1, "decider joined more than one deal");
+        deals.remove(0)
+    };
+
+    // Check that we can find worker_id by deal_id
+    let resolved_worker_id = {
+        let mut result = execute(
+            &mut client,
+            r#"(call relay ("worker" "get_worker_id") [deal_id] resolved_worker_id)"#,
+            "resolved_worker_id",
+            hashmap! {
+                "deal_id" => json!(deal.deal_id),
+            },
+        )
+        .await
+        .unwrap();
+        let mut worker_id = serde_json::from_value::<Vec<String>>(result.remove(0)).unwrap();
+        assert!(!worker_id.is_empty(), "can't resolve worker_id by deal_id");
+        worker_id.remove(0)
+    };
+
+    assert_eq!(deal.worker_id, resolved_worker_id);
+    assert_eq!(deal.deal_id, format!("0x{DEAL_ID}"));
+
+    let state = get_deal_state(&mut client, &deal.deal_id).await;
+    assert_eq!(
+        state.left_boundary,
+        to_hex(BLOCK),
+        "wrong saved state for the deal"
+    );
+
+    // Check that we can see the list of services on the worker
+
+    #[derive(Deserialize, Debug)]
+    struct ServiceInfo {
+        aliases: Vec<String>,
+        worker_id: String,
+        service_type: String,
+    }
+
+    // Here we also test that the Installation Spell worked correctly to ensure that the distro is fine,
+    // but deep Installation Spell testing is out of scope of this test suits
+    wait_worker_spell(
+        &mut client,
+        deal.worker_id.clone(),
+        Duration::from_millis(200),
+    )
+    .await;
+
+    let worker_service_list = {
+        let result = execute(
+            &mut client,
+            r#"
+            (seq
+                (call relay ("op" "noop") [])
+                (call worker_id ("srv" "list") [] list)
+            )
+        "#,
+            "list",
+            hashmap! {
+                "worker_id" => json!(deal.worker_id),
+            },
+        )
+        .await
+        .unwrap();
+        serde_json::from_value::<Vec<ServiceInfo>>(result[0].clone()).unwrap()
+    };
+
+    let test_app_1 = TestApp::test_app1();
+    let worker_spell = worker_service_list
+        .iter()
+        .find(|info| info.aliases.contains(&"worker-spell".to_string()));
```

Comment:
`& .to_string()` looks strange, sure it won't take `&'static str`?

Reply:
No, `contains` wants `&String